icu4c/source/i18n/number_longnames.cpp

```diff
@@ -295,17 +440,47 @@ void LongNameHandler::forCompoundUnit(const Locale &loc, const MeasureUnit &unit
         status = U_INTERNAL_PROGRAM_ERROR;
         return;
     }
+
+    StringPiece primaryCase, secondaryCase;
+    CharString primaryCase_, secondaryCase_;
```

Comment:
use the `variable_` for internal class variables, I kind feel it is confusing here.

Reply:
Thanks for catching! Agreed - bad names. I've renamed them to *Mem to indicate they're in charge of holding memory. 

FWIW I took inspiration from this one:
```
static const MeasureUnitImpl& MeasureUnitImpl::forMeasureUnit(
        const MeasureUnit& measureUnit, MeasureUnitImpl& memory, UErrorCode& status);
```